File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int main(void)
{
    string text = get_string("Text: ");
    float num_letters = count_letters(text);
    float num_words = count_words(text);
    float num_sentences = count_sentences(text);
    float L = (num_letters * 100) / num_words;
    float S = (num_sentences * 100) / num_words;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int index_round = round(index);
    if (index_round < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index_round >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index_round);
    }
}

int count_letters(string text)
{
    int sum = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            sum++;
        }
    }
    return sum;
}

int count_words(string text)
{
    int sum = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]))
        {
            if (!isspace(text[i - 1]))
            {
                sum++;
            }
        }
    }
    return sum + 1;
}

int count_sentences(string text)
{
    int sum = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sum++;
        }
    }
    return sum;
}

File: sentimental-readability/readability.py
from cs50 import *

text = get_string("Text:")

words = 1
letters = 0
sentences = 0

for c in text:
    if c.isalpha():
        letters += 1
    elif c.isspace():
        words += 1
    elif c == "." or c == "?" or c == "!":
        sentences += 1

L = float(letters) / float(words) * 100
S = float(sentences) / float(words) * 100

index = round(0.0588 * L - 0.296 * S - 15.8)

if index >= 16:
    print("Grade 16+")

elif index < 1:
    print("Before Grade 1")

else:
    print("Grade", index)
